lib/pf/util.pm

```diff
@@ -1011,6 +1012,28 @@ sub cert_has_expired {
     return time > $expiration;
 }
 
+=item strip_username
+
+Will strip a username matching pattern user@realm or \\realm\user
+
+Returns ($user,$realm) if found or ($user) if not matching any realm pattern
+
+=cut
+
+sub strip_username {
+    my ($username) = @_;
+    if($username =~ /(.*)\@(.*)/){
+        return ($1,$2);
+    }
+    elsif($username =~ /(.*)\%(.*)/){
+        return ($1,$2);
+    }
+    elsif($username =~ /\\\\(.*)\\(.*)/) {
+        return ($2,$1);
+    }
+    return $username;
+}
+
```

Comment:
Don't think we need named regex but having a simple line comment describing which use case we are matching would be great.